libs/server/SessionParseStateExtensions.cs

```diff
@@ -0,0 +1,163 @@
+        public static bool TryGetClientType(this SessionParseState parseState, int idx, out ClientType value)
+        {
+            value = ClientType.Invalid;
+            var sbArg = parseState.GetArgSliceByRef(idx).ReadOnlySpan;
+
+            if (sbArg.EqualsUpperCaseSpanIgnoringCase("NORMAL"u8))
+                value = ClientType.NORMAL;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("MASTER"u8))
+                value = ClientType.MASTER;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("REPLICA"u8))
+                value = ClientType.REPLICA;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("PUBSUB"u8))
+                value = ClientType.PUBSUB;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("SLAVE"u8))
+                value = ClientType.SLAVE;
+
+            return value != ClientType.Invalid;
+        }
+
+        /// <summary>
+        /// Parse bit field overflow from parse state at specified index
+        /// </summary>
+        /// <param name="parseState">The parse state</param>
+        /// <param name="idx">The argument index</param>
+        /// <param name="value">Parsed value</param>
+        /// <returns>True if value parsed successfully</returns>
+        internal static bool TryGetBitFieldOverflow(this SessionParseState parseState, int idx, out BitFieldOverflow value)
+        {
+            value = default;
+            var sbArg = parseState.GetArgSliceByRef(idx).ReadOnlySpan;
+
+            if (sbArg.EqualsUpperCaseSpanIgnoringCase("WRAP"u8))
+                value = BitFieldOverflow.WRAP;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("SAT"u8))
+                value = BitFieldOverflow.SAT;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("FAIL"u8))
+                value = BitFieldOverflow.FAIL;
+            else return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Parse manager type from parse state at specified index
+        /// </summary>
+        /// <param name="parseState">The parse state</param>
+        /// <param name="idx">The argument index</param>
+        /// <param name="value">Parsed value</param>
+        /// <returns>True if value parsed successfully</returns>
+        internal static bool TryGetManagerType(this SessionParseState parseState, int idx, out ManagerType value)
+        {
+            value = default;
+            var sbArg = parseState.GetArgSliceByRef(idx).ReadOnlySpan;
+
+            if (sbArg.EqualsUpperCaseSpanIgnoringCase("MIGRATIONMANAGER"u8))
+                value = ManagerType.MigrationManager;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("REPLICATIONMANAGER"u8))
+                value = ManagerType.ReplicationManager;
+            else if (sbArg.EqualsUpperCaseSpanIgnoringCase("SERVERLISTENER"u8))
+                value = ManagerType.ServerListener;
+            else return false;
+
+            return true;
+        }
+    }
+}
```

Comment:
What about the enums we handled manually earlier, such as `TryGetSortedSetAddOption`?

Reply:
Added these 2, didn't find any others...